src/app/core/authentication/login.service.ts

```diff
@@ -29,4 +35,8 @@ export class LoginService {
   menu() {
     return this.http.get<{ menu: Menu[] }>('/me/menu').pipe(map(res => res.menu));
   }
+
+  setDefaults(user: User) {
+    return Object.assign(this.defaults, user);
+  }
 }
```

Comment:
I think it‘s more convenient to set default user in `user.ts` and the `login.service.ts` should only have APIs.

Reply:
I  add Config and ConfigService, the developer can use AUTH_CONFIG to set API Url and user default value